docs/elasticsearch/bool.md

```diff
@@ -0,0 +1,206 @@
+          }
+        }
+      ],
+      "should": [
+        {
+          "match": {
+            "text_entry": "life"
+          }
+        },
+        {
+          "match": {
+            "text_entry": "grace"
+          }
+        }
+      ],
+      "minimum_should_match": 1,
+      "must_not": [
+        {
+          "match": {
+            "speaker": "ROMEO"
+          }
+        }
+      ],
+      "filter": {
+        "match": {
+          "play_name": "Romeo and Juliet"
+        }
+      }
+    }
+  }
+}
+```
+
+#### Sample output
+
+```json
+{
+  "took": 12,
+  "timed_out": false,
+  "_shards": {
+    "total": 4,
+    "successful": 4,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 11.356054,
+    "hits": [
+      {
+        "_index": "shakespeare",
+        "_type": "_doc",
+        "_id": "88020",
+        "_score": 11.356054,
+        "_source": {
+          "type": "line",
+          "line_id": 88021,
+          "play_name": "Romeo and Juliet",
+          "speech_number": 19,
+          "line_number": "4.5.61",
+          "speaker": "PARIS",
+          "text_entry": "O love! O life! not life, but love in death!"
+        }
+      }
+    ]
+  }
+}
+```
+
+If you want to identify which of these clauses actually caused the matching results, name each of your queries with the `_name` parameter.
+To add the `_name` parameter, change the field name in the `match` query to an object:
+
+
+```json
+GET shakespeare/_search
+{
+  "query": {
+    "bool": {
+      "must": [
+        {
+          "match": {
+            "text_entry": {
+              "query": "love",
+              "_name": "love-must"
+            }
+          }
+        }
+      ],
+      "should": [
+        {
+          "match": {
+            "text_entry": {
+              "query": "life",
+              "_name": "life-should"
+            }
+          }
+        },
+        {
+          "match": {
+            "text_entry": {
+              "query": "grace",
+              "_name": "grace-should"
+            }
+          }
+        }
+      ],
+      "minimum_should_match": 1,
+      "must_not": [
+        {
+          "match": {
+            "speaker": {
+              "query": "ROMEO",
+              "_name": "ROMEO-must-not"
+            }
+          }
+        }
+      ],
+      "filter": {
+        "match": {
+          "play_name": {
+            "query": "Romeo and Juliet",
+            "_name": "Romeo-and-Juliet-filter"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+You get back a `matched_queries` array that lists the queries that matched these results:
+
+```json
+"matched_queries": [
+  "Romeo-and-Juliet-filter",
+  "love-must",
+  "life-should"
+]
+```
+
+If you remove the queries not in this list, you will still see the exact same result.
+By examining which `should` clause matched, you can better understand the relevancy score of the results.
```

Comment:
Ending feels a little abrupt. Any additional detail we should add here, or this concludes it?

Reply:
Yes, it does. I added an example for nested bool queries at the end.